raft/raft.go

```diff
@@ -448,6 +452,12 @@ func (r *raft) tickHeartbeat() {
 		r.heartbeatElapsed = 0
 		r.Step(pb.Message{From: r.id, Type: pb.MsgBeat})
 	}
+
+	// If current leader cannot transfer leader in election timeout, become leader again.
+	// As per raft thesis 3.10
+	if r.state == StateTransferLeader {
+		r.becomeLeader()
```

Comment:
I think `becomeLeader` does too much: scanning the tail of the log for uncommitted config changes, appending a dummy entry, etc. All we really need here is to reset a few fields including `r.state` and `r.step`, then continue as if our leadership was never interrupted.